RingBufCPP.h

```diff
@@ -89,6 +166,23 @@ Type* peek(size_t num)
     return ret;
 }
 
+/**
+* Peek at oldest element in the buffer.
+* Return: a pointer to the element (nullptr if buffer is empty)
+*/
+Type* peek() {
```

Comment:
Isn't this equivalent to `peek(size_t num=0)` (i.e. just with a default argument)?

Reply:
Yes it's the same but it's faster because you can spare two operations (sum and division) (... + num) % MaxElements];)
I know I'm being picky, but we are thinking "embedded" here. And a division could be a big deal on a microcontroller. Normally most of the users will want to peek for the first element...

Reply:
MaxElements is a static compile time parameter. If you make it a power of 2 the compiler will absolutely use bit shift operations instead of raw division operations. For almost every platform this would be a single instruction. Unless you have profiling data showing this optimization helps for power of 2 sized buffers, I would prefer we just do this with default arguments.